packages/contracts-bedrock/src/L1/DataAvailabilityChallenge.sol

```diff
@@ -0,0 +1,419 @@
+        // require the commitment type to be known
+        CommitmentType commitmentType = _validateCommitment(challengedCommitment);
+
+        // require the challenge to be active (started, not resolved, and resolve window still open)
+        if (getChallengeStatus(challengedBlockNumber, challengedCommitment) != ChallengeStatus.Active) {
+            revert ChallengeNotActive();
+        }
+
+        // compute the commitment corresponding to the given resolveData
+        bytes memory computedCommitment;
+        if (commitmentType == CommitmentType.Keccak256) {
+            computedCommitment = computeCommitmentKeccak256(resolveData);
+        }
+
+        // require the provided input data to correspond to the challenged commitment
+        if (keccak256(computedCommitment) != keccak256(challengedCommitment)) {
+            revert InvalidInputData(computedCommitment, challengedCommitment);
+        }
+
+        // store the block number at which the challenge was resolved
+        Challenge storage activeChallenge = challenges[challengedBlockNumber][challengedCommitment];
+        activeChallenge.resolvedBlock = block.number;
+
+        // emit an event to notify that the challenge status is now resolved
+        emit ChallengeStatusChanged(challengedBlockNumber, challengedCommitment, ChallengeStatus.Resolved);
+
+        // distribute the bond among challenger, resolver and address(0)
+        _distributeBond(activeChallenge, resolveData.length, msg.sender);
+    }
+
+    /// @notice Distribute the bond of a resolved challenge among the resolver, challenger and address(0).
+    ///         The challenger is refunded the bond amount exceeding the resolution cost.
+    ///         The resolver is refunded a percentage of the resolution cost based on the `resolverRefundPercentage`
+    ///         state variable.
+    ///         The remaining bond is burned by sending it to address(0).
+    /// @dev The resolution cost is approximated based on a fixed cost and variable cost depending on the size of the
+    ///      pre-image.
+    ///      The real resolution cost might vary, because calldata is priced differently for zero and non-zero bytes.
+    ///      Computing the exact cost adds too much gas overhead to be worth the tradeoff.
+    /// @param resolvedChallenge The resolved challenge in storage.
+    /// @param preImageLength The size of the pre-image used to resolve the challenge.
+    /// @param resolver The address of the resolver.
+    function _distributeBond(Challenge storage resolvedChallenge, uint256 preImageLength, address resolver) internal {
+        uint256 lockedBond = resolvedChallenge.lockedBond;
+        address challenger = resolvedChallenge.challenger;
+
+        // approximate the cost of resolving a challenge with the provided pre-image size
+        uint256 resolutionCost = (
+            fixedResolutionCost + preImageLength * variableResolutionCost / variableResolutionCostPrecision
+        ) * tx.gasprice;
```

Comment:
I am still a little suspect of `ts.gasprice` and would feel a lot better if there was something along the lines of:

```solidity
if (block.basefee > tx.gasprice) revert
```

I know 1559 should enforce this in the protocol but idk if a new tx type would change this invariant somehow

Reply:
had a comment about that here: https://github.com/ethereum-optimism/optimism/pull/9356#discussion_r1492933690

Reply:
I'm fine with changing the calculation to use the basefee instead though (as per https://github.com/ethereum-optimism/optimism/pull/9356#discussion_r1483136616 and https://github.com/ethereum-optimism/optimism/pull/9356#discussion_r1498297246)

Reply:
changed to `block.basefee`